dataflux_pytorch/benchmark/checkpointing/singlenode/train.py

```diff
@@ -132,26 +134,52 @@ def main():
     )
     trainer.fit(model, dataloader)
 
-    start = time.time()
-    for i in range(args.steps):
-        trainer.save_checkpoint(
-            os.path.join(args.ckpt_dir_path, f'ckpt_{i}.ckpt'))
-    end = time.time()
-    # command to clear kernel cache only works on MacOs and Linux.
-    if args.clear_kernel_cache and sys.platform in [
-            "darwin", "linux", "linux2", "linux3"
-    ]:
-        print("Clearing kernel cache...")
-        os.system("sync && sudo sysctl -w vm.drop_caches=3")
-    print("Average time to save one checkpoint: " +
-          str((end - start) / args.steps) + " seconds")
-    start = time.time()
-    for i in range(args.steps):
-        data = ckpt.load_checkpoint(
-            os.path.join(args.ckpt_dir_path, f'ckpt_{i}.ckpt'))
-    end = time.time()
-    print("Average time to load one checkpoint: " +
-          str((end - start) / args.steps) + " seconds")
+    if args.no_lightning:
```

Comment:
Would it be possible to reduce the code duplication at all? One way might be to put if statements only inside each of the `for i in range(args.steps):` loops that determine how exactly the checkpoint is saved/loaded